devel/create-github-release

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -euo pipefail
+shopt -s extglob
+
+devel="$(dirname "$0")"
+
+main() {
+    local version="${1:?version is required}"
+    local commit pre_release
+    shift
+    local -a assets=("$@")
+
+    if [[ ${#assets[@]} -eq 0 ]]; then
+        echo "ERROR: no assets provided" >&2
+        exit 1
+    fi
+
+    # Asserts that $version is an actual tag, not a branch name or other ref.
+    #
+    # Prefixing with refs/tags/ is a bit hacky and suffixing with ^{tag} would
+    # be more proper to restrict to a proper annotated, meant-for-release tag.
+    # While this is dandy in any given clone, in the GitHub Actions context the
+    # annotated tag is intentionally **overwritten** and converted to a
+    # lightweight tag by actions/checkout@v3, which doesn't/can't pass the
+    # muster of ^{tag}.
+    git rev-parse --verify "refs/tags/$version" >/dev/null
+
+    # Translate from tag into commit for `gh release create --target`.
+    commit="$(git rev-parse --verify "$version^{commit}")"
+
+    if is-pre-release "$version"; then
+        pre_release=1
+    fi
+
+    gh release create \
+        "$version" \
+        --repo "${GITHUB_REPOSITORY:-nextstrain/cli}" \
+        --title "$version" \
+        --target "$commit" \
```

Comment:
Wait, is `--target` even necessary since `"$version"` is already provided? I tried running locally without it and seems to work fine.

Reply:
From [the `gh` docs](https://cli.github.com/manual/gh_release_create):

> If a matching git tag does not yet exist, one will automatically get created from the latest state of the default branch. Use `--target` to override this.

Reply:
I see. Would be nice to have `--target` accept a tag or `refs/tags/<tag>` but seems like that's not currently supported (https://github.com/cli/cli/issues/2133#issuecomment-1164899636).

Reply:
Agreed! I'd tried with a tag but it didn't work, and was unsurprised since the doc says it must be branch name or full commit id. On the other hand, the error message from the API refers to the `target_commitish` field, and a tag is a commit-ish in git parlance (see `man gitglossary`), so from that name you'd expect it to work!